src/config/config.go

```diff
@@ -1,7 +1,145 @@
 package config
 
-const InsertDefaultRank bool = true
-const LogDumpLocation string = "logdump/"
+import (
+	"os"
+	"strings"
 
-// update this when loading config
-var NumberOfEngines int = 10
+	"github.com/knadh/koanf/parsers/yaml"
+	"github.com/knadh/koanf/providers/env"
+	"github.com/knadh/koanf/providers/file"
+	"github.com/knadh/koanf/providers/structs"
+	"github.com/knadh/koanf/v2"
+	"github.com/rs/zerolog/log"
+	"github.com/tminaorg/brzaguza/src/structures"
+)
+
+type SESettings struct {
+	Timeout  int                 `koanf:"timeout"`
+	Shortcut string              `koanf:"shortcut"`
+	Crawlers []structures.Engine `koanf:"crawlers"`
+}
+
+type Engine struct {
+	Enabled  bool       `koanf:"enabled"`
+	Settings SESettings `koanf:"settings"`
+}
+
+// Config struct for Koanf
+type Config struct {
+	Engines map[structures.Engine]Engine `koanf:"engines"`
+}
```

Comment:
Is the idea to map multiple engine strings (names (structures.Engine)) to the same config.Engine variable? Then it should be a map to *Engine, no? Is the mapping (the actual links, not talking about the settings themselves) supposed to be modifiable at runtime? I can't think of a valid reason why they should be. This shouldn't be a map.

Reply:
It can be a list of objects (Engines) with added field `name`. Config would look like this in that case:
```yaml
engines:
  - name: google
    enabled: false
  - name: duckduckgo
    enabled: true
```
instead of the original with map:
```yaml
engines:
  google:
    enabled: false
  duckduckgo:
    enabled: true
 ```